File: src/main/scala/test/Person.scala
package test

/**
  * [1] 直接在类名后面跟参数为主构造函数;
  * [2] 可以通过this定义辅助构造函数
  *
  * @param n
  * @param a
  */
class Person(n: String, a: Int) {

  var name: String = n
  var age = a

  def this(a: Int) {
    this("", a)
  }

  override def toString: String = "name: " + name + " age：" + a

}

File: src/main/java/sparksqljava/teacher/Person.java
package sparksqljava.teacher;

import java.io.Serializable;

public class Person implements Serializable{
	private String name;
	private Long age;
	private String address;
	public Person(){
	}
	public Person(String name, Long age, String address) {
		super();
		this.name = name;
		this.age = age;
		this.address = address;
	}
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public Long getAge() {
		return age;
	}
	public void setAge(Long age) {
		this.age = age;
	}
	public String getAddress() {
		return address;
	}
	public void setAddress(String address) {
		this.address = address;
	}
}

